lib/portlayer/exec/container.go

```diff
@@ -315,9 +316,10 @@ func (c *Container) WaitForState(s State) <-chan struct{} {
 func (c *Container) NewHandle(ctx context.Context) *Handle {
```

Comment:
You have the freedom to pass an operation and if you do it will use that operation...otherwise create a new one.  I didn't change the signature in an effort to limit the changes..